lib/friendly_id/finder_methods.rb

```diff
@@ -31,7 +31,9 @@ def find(*args, allow_nil: false)
       first_by_friendly_id(id).tap { |result| return result unless result.nil? }
       return super(*args) if potential_primary_key?(id)
 
-      raise_not_found_exception(id) unless allow_nil
+      raise_not_found_exception(id)
+    rescue => e
+      allow_nil ? nil : raise(e)
```

Comment:
```suggestion
      raise_not_found_exception(id) unless allow_nil
    rescue ActiveRecord::RecordNotFound => exception
      raise exception unless allow_nil
```

What about this version? If we avoid raising the exception in the first place then we get higher performance by avoiding a needless raise/rescue/raise cycle.